src/components/user/handlers/user-project-member-roles.handler.ts

```diff
@@ -0,0 +1,45 @@
+import { difference } from 'lodash';
+import {
+  DatabaseService,
+  EventsHandler,
+  IEventHandler,
+  ILogger,
+  Logger,
+} from '../../../core';
+import { AuthorizationService } from '../../authorization/authorization.service';
+import { ProjectMemberUpdatedEvent, UserUpdatedEvent } from '../events';
+
+type SubscribedEvent = UserUpdatedEvent | ProjectMemberUpdatedEvent;
+
+@EventsHandler(UserUpdatedEvent, ProjectMemberUpdatedEvent)
```

Comment:
As written, when a project member loses a role, this will delete them from the same role grant on any project. This handler shouldn't handle the `ProjectMemberUpdatedEvent`. We need to have a specific handler for that event that only removes the user from the role(s) on the project.

So there are two scenarios this PR needs to cover:
1. A user's role is removed from their user node. This is a global removal of a role. They are no longer authorized to have this role anywhere. The apoc call currently in this PR does that.
2. A user's role is removed on their project member node on a project. In this case, we just need to remove their membership on the security group for their role on that one project.